app/models/derived_variables/case_log_variables.rb

```diff
@@ -186,4 +187,23 @@ def reset_scheme_location!
       self.location = scheme.locations.first
     end
   end
+
+  def set_housingneeds_fields
+    self.housingneeds_a = fully_wheelchair_accessible? ? 1 : 0
+    self.housingneeds_b = essential_wheelchair_access? ? 1 : 0
+    self.housingneeds_c = level_access_housing? ? 1 : 0
+    self.housingneeds_f = other_housingneeds? ? 1 : 0
+    set_housingneeds_values_to_zero unless has_housingneeds?
```

Comment:
It feels brittle that this is order dependent but I suppose that's already the case for a whole bunch of derivation

Reply:
Think it might work better inline, as a single method that sets housing needs fields but just personal preference so don't feel the need to change